src/datetime_queries/fetch_in_time_range.rs

```diff
@@ -0,0 +1,301 @@
+            &mock_fetchers,
+            start_time,
+            end_time,
+            base_component.clone(),
+        );
+        assert_eq!(result, Ok(wire_vec.clone()));
+        // ==================================================
+        // hour to day test
+        // ==================================================
+
+        let mock_day_to_day = fetch_entries_from_day_to_day::MockFetchByDayDay::new();
+        let mock_day_to_hour = fetch_entries_from_day_to_hour::MockFetchByDayHour::new();
+        let mock_hour_to_hour = fetch_entries_from_hour_to_hour::MockFetchByHourHour::new();
+        let mock_by_day = fetch_by_day::MockFetchByDay::new();
+        let mock_by_hour = fetch_by_hour::MockFetchByHour::new();
+        let mock_get_latest = get_latest_for_entry::MockGetLatestEntry::new();
+        // create inputs and outputs
+        let start_time = FetchEntriesTime {
+            year: 2021,
+            month: 10 as u32,
+            day: 20 as u32,
+            hour: Some(10 as u32),
+        };
+        let end_time = FetchEntriesTime {
+            year: 2021,
+            month: 10 as u32,
+            day: 21 as u32,
+            hour: None,
+        };
+        let mut mock_hour_to_day = fetch_entries_from_hour_to_day::MockFetchByHourDay::new();
+        mock_hour_to_day
+            .expect_fetch_entries_from_hour_to_day::<Example>()
+            .with(
+                mockall::predicate::always(),
+                mockall::predicate::eq(start_time.clone()),
+                mockall::predicate::eq(end_time.clone()),
+                mockall::predicate::eq(base_component.clone()),
+            )
+            .times(1)
+            .return_const(Ok(wire_vec.clone()));
+
+        let mock_fetchers = Fetchers {
+            day_to_day: mock_day_to_day,
+            day_to_hour: mock_day_to_hour,
+            hour_to_day: mock_hour_to_day,
+            hour_to_hour: mock_hour_to_hour,
+            day: mock_by_day,
+            hour: mock_by_hour,
+            get_latest: mock_get_latest,
+        };
+
+        let result = super::fetch_entries_in_time_range::<Example>(
+            &mock_fetchers,
+            start_time,
+            end_time,
+            base_component.clone(),
+        );
+        assert_eq!(result, Ok(wire_vec.clone()));
+        // ==================================================
+        // hour to day test
+        // ==================================================
+
+        let mock_day_to_day = fetch_entries_from_day_to_day::MockFetchByDayDay::new();
+        let mock_day_to_hour = fetch_entries_from_day_to_hour::MockFetchByDayHour::new();
+        let mock_hour_to_day = fetch_entries_from_hour_to_day::MockFetchByHourDay::new();
+        let mock_by_day = fetch_by_day::MockFetchByDay::new();
+        let mock_by_hour = fetch_by_hour::MockFetchByHour::new();
+        let mock_get_latest = get_latest_for_entry::MockGetLatestEntry::new();
```

Comment:
pare this back to one line once you have a constructor for `Fetchers`. 
Something like
`let mut fetchers = Fetchers::new()`
then you set up your expectations

same thing with all similar instances

Reply:
okay so the constructor doesn't take any input and assumes each property is to be filled with the mock struct?

Reply:
The mock or not the mock depending on mode